packages/smithy-client/src/client.ts

```diff
@@ -87,6 +113,7 @@ export class Client<
   }
 
   destroy() {
-    if (this.config.requestHandler.destroy) this.config.requestHandler.destroy();
+    this.config.requestHandler.destroy?.();
+    clientHandlers.delete(this);
```

Comment:
if the concern is affecting the shape of the class, we can probably use `private handler?` i.e. optionality in the property, for backwards compatibility.